### PR TITLE
Add constructor in Peak that allows converting from LeanPeak to Peak (SCD196)

### LINKED ISSUE #30867: Convert a LeanPeak into a standard Peak
We should add methods on Peak and LeanPeak to convert between the two. This is easy for Peak to LeanPeak but for LeanPeak to Peak we will require that an instrument be provided.
Or implement a constructor that takes the other type as imput.

> NOTE: This issue is originally task [SCD196](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/196) on Gitlab.

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**

This PR add a new constructor to Peak class which can convert a LeanElasticPeak to Peak.

- Related Gitlab task: [SCD196](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/196)
- This is PR #30866 into `master`

**To test:**
Build Mantid and run unittest `ctest -R PeakTest`.

Fixes #30867 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataObjects/src/Peak.cpp [hunk 1]
```diff
@@ -177,6 +178,14 @@ Peak::Peak(const Geometry::IPeak &ipeak)
   }
 }
 
+//----------------------------------------------------------------------------------------------
+Peak::Peak(const Mantid::DataObjects::LeanElasticPeak &lpeak,
+           const Geometry::Instrument_const_sptr &inst,
+           boost::optional<double> detectorDistance) {
```
Comment (rosswhitfield) [inline Framework/DataObjects/src/Peak.cpp:184]: Can you add in `: BasePeak(lpeak)` in order to copy over all the member variables in the `BasePeak`

File: Framework/DataObjects/test/PeakTest.h [hunk 2]
```diff
@@ -123,6 +123,12 @@ class PeakTest : public CxxTest::TestSuite {
     check_Contributing_Detectors(p2, expectedIDs);
   }
 
+  void test_ConstructorFromLeanElasticPeak() {
+    const LeanElasticPeak &lpeak = LeanElasticPeak(V3D(1, 2, 3));
```
Comment (rosswhitfield) [inline Framework/DataObjects/test/PeakTest.h:127]: Can you have an example using a goniometer that is not identity. Then actually check that the `.getQLabFrame` and `.getQSampleFrame` give the same results between Peak and LeanElasticPeak.

**Review by rosswhitfield — CHANGES_REQUESTED** [review]
If you look at the test for going from Peak to LeanElasticPeak [here](https://github.com/mantidproject/mantid/blob/36e36bdc1e118c2f972b18333a9d296ac8542db4/Framework/DataObjects/test/LeanElasticPeakTest.h#L261) you should be able to go the other direction and be able to check all the same properties.

File: Framework/DataObjects/src/Peak.cpp (hunk 1 above)
Comment (KedoKudo) [reply Framework/DataObjects/src/Peak.cpp]: Thanks for pointing it out.  The constructor has been updated accordingly.

File: Framework/DataObjects/test/PeakTest.h (hunk 2 above)
Comment (KedoKudo) [reply Framework/DataObjects/test/PeakTest.h]: More robust unit test following the example in LeanElasticPeakTest has been implemented.

**Review by rosswhitfield — APPROVED** [review]

**Pull request merged**